lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -12,6 +13,55 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     div(slot, slots_per_epoch)
   end
 
+  @doc """
+     Return the shuffled index corresponding to ``seed`` (and ``index_count``).
+  """
+  @spec compute_shuffled_index(SszTypes.uint64(), SszTypes.uint64(), SszTypes.bytes32()) ::
+          {:error, String.t()}
+  def compute_shuffled_index(index, index_count, _seed)
+      when index >= index_count or index_count == 0 do
+    {:error, "invalid index_count"}
+  end
+
+  @spec compute_shuffled_index(SszTypes.uint64(), SszTypes.uint64(), SszTypes.bytes32()) ::
+          {:ok, SszTypes.uint64()}
+  def compute_shuffled_index(index, index_count, seed) do
+    shuffle_round_count = ChainSpec.get("SHUFFLE_ROUND_COUNT")
+
+    new_index =
+      Enum.reduce(0..(shuffle_round_count - 1), index, fn round, current_index ->
+        round_as_bytes = <<round>>
+
+        hash_of_seed_round = :crypto.hash(:sha256, seed <> round_as_bytes)
+
+        pivot = rem(bytes_to_uint64(hash_of_seed_round), index_count)
+
+        flip = rem(pivot + index_count - current_index, index_count)
+        position = max(current_index, flip)
+
+        position_div_256 = uint_to_bytes4(div(position, 256))
+
+        source =
+          :crypto.hash(:sha256, seed <> round_as_bytes <> position_div_256)
+
+        byte_index = div(rem(position, 256), 8)
+        byte = source |> :binary.bin_to_list() |> Enum.fetch!(byte_index)
```

Comment:
We can use pattern matching here
```suggestion
        <<_::binary-size(byte_index - 1), byte, _::binary>> = source
```
(this matches the first `byte_index - 1` bytes, binds the next byte to `byte`, and discards the rest)

Reply:
Great! Had to change `byte_index - 1` to `byte_index`